### PR TITLE
[Types] Fix the sort's object notation (entity service)

### LINKED ISSUE #18263: Populate generated types doesn't work with filter on entityService
### Required System information

- Node.js version: v16.20.0
- NPM version: 8.19.4
- Strapi version: 4.13.7
- Database: postfgres
- Operating system: wsl
- Is your project Javascript or Typescript: Typescript

### Describe the bug

I'm using entityService with the generated types, typing my strapi, and I note when a add a filter on my api, the populate fields type are gone from my types.

Before add filters (without error):
image

On add filters (with type error):
image

### Steps to reproduce the behavior

1. Creata a collection type with relation
2. Create a api with entityService without filters and populate and check the types
3. Add filters and check the types

### Expected behavior

Don't gone my populated fields from types.

### Additional context

I don't know if is a bug or I'm using wrong, if anyone can help, thanks.

### PR DESCRIPTION
**Pull request by Convly**
### What does it do?

**tl;dr: Refactored the sort's object notation to fix inference and index signatures conflicts**

- Moved away from an intersection of mapped types (with different values) as their respective index signatures were conflicting.
- Used a single mapped type
	- It uses a union of every sortable key (populatable and non-populatable) as index
	- The value type is assigned directly based on the key type (using conditionals)

### Why is it needed?

Currently, it is not possible to use a combination of `fields`/`populate` and `sort` params (using the sort's object notation), as it'll break the params inference.

This results in undesired behavior in users' code (e.g. #18263)

### How to test it?

Using any Strapi project with generated types
- Go into a file that has access to both generated types and the `strapi` global object
- Write an `entityService.findMany` query for a collection-type.
- Add a populate attribute to hand-pick some relations or component
- The `findMany`'s result should propose only the populatable fields that have been selected
- Now, add a sort param using the object notation, and sort on any field available
- **The `findMany` result should still propose populated fields** 

### Related issue(s)/PR(s)

fix #18263

___

_also, fixed the watch script for the core/utils package_

### CODE REVIEW

**Comment by Convly** [comment]
Experimental release available: `0.0.0-experimental.16a54daf6a3e687d54fda4065a349f30931498f4`

**Comment by Convly** [comment]
The issue with sort has been fixed, but it appears filters are also triggering issues. Flagging as `don't merge` while I'm working on this.

**Comment by Convly** [comment]
**Update**: After battling with TS, I think I've finally made a breakthrough and issued a fix that should solve the bug. I'm still doing some research on the side to understand what's happening exactly, but I'll update the PR soon.
